### DISCUSSION TITLE
Note Creation: Strange Behaviour?

### DISCUSSION THREAD

**Discussion by proteus-dev** [DeabLabs/cannoli#10] [Q&A]
Hi, 

Thanks again for Cannoli!
I am currently using it for cleaning imported websites (via MarkDownload Chrome Plugin) and creating summaries, which works like a charm! ...initially...
My problem is after re-arranging some nodes in the canvas, my created cannoli does not write the expected output but the path into the newly created note (see attached screenshot).
I reconstructed my canvas several times from the Cannoli College template (4.4). It appears as if it always works one or two times (sorry, I am tired...) as expected.

Any suggestions? :)

<img width="1325" alt="image" src="https://github.com/DeabLabs/cannoli/assets/121372604/6020ca6c-9f10-464f-a17b-5509ccc16c0d">

**Comment by blindmansion** [comment]
Thanks for raising this. When I get a chance this week ill try to reproduce this and see if I can figure out what's happening.

To clarify, you're saying that when running the same cannoli several times, the first few times it works, then it starts writing the path to the new note instead of the intended content? Is the note being created at the correct path still?

My initial guess is that the path is being found to be invalid or something, but I'm not sure. Ill give an update when I know more.

**Reply by proteus-dev** [reply]
yes, after a few times it starts writing the path.
notes are created and enumerated in case of duplicate titles.

**Reply by proteus-dev** [reply]
maybe it is about the title and the long dash "—". I will try tomorrow.

**Reply by proteus-dev** [reply]
I did the following test today:
- fail: delete all existing files related to note creation
- fail: make sure that only alphabetical characters are in the title
- fail: create new canvas, copy cannoli college 4.4 and insert my file and prompts

Yesterday I wrote a debugger to see how each step creates an output. Seems like the output is fine until the last user step.

<img width="951" alt="image" src="https://github.com/DeabLabs/cannoli/assets/121372604/2da0d3a3-4fd7-4c83-a0a2-f1c17a2b5c4d">

**Reply by blindmansion** [reply]
I reproduced this and I think I've identified the issue! I still need to test it and make sure my fix doesn't cause any regressions. I will try to release the fix within the week.

Thanks for discovering this! You are definitely cannoli's preeminent power user, excited to see what else you build, and what other arcane bugs you root out.

**Reply by blindmansion** [reply]
So sorry for the delay on this, it should be fixed with the most recent release! Let me know if it works as expected.
